In [ ]:
## Notebook 02: Artists & Tracks

**Purpose:**  
Explore artist- and track-level listening behavior to understand concentration, repetition, and preference patterns.

**Key Questions:**
- Which artists and tracks dominate listening by play count?
- How do rankings change when measured by total listening time?
- Is listening concentrated among a small set of artists, or spread broadly?

**Constraints:**
- Analysis uses cleaned Parquet data (read-only)
- Track and artist fields are treated as labels, not stable Spotify IDs
- No session logic in this notebook

**Outputs:**
- 2–3 labeled charts
- Written insight bullets per chart

In [1]:
import sys
sys.executable

'D:\\PythonVenvs\\spotify-listening-analytics-env\\.venv\\Scripts\\python.exe'

In [2]:
import pandas as pd
from pathlib import Path


In [3]:
DATA_PATH = Path("../data/cleaned/listening_cleaned.parquet")
DATA_PATH.exists()

True

In [4]:
df = pd.read_parquet(DATA_PATH)
df.shape

(66098, 9)

In [5]:
list(df.columns)

['timestamp_utc',
 'utc_time',
 'artist',
 'artist_mbid',
 'album',
 'album_mbid',
 'track_id',
 'track_mbid',
 'timestamp_local']

### Analysis Plan

1. Rank artists by number of plays
2. Rank artists by total listening time (if available)
3. Compare concentration between the two metrics
4. Repeat the same analysis for tracks (top-N only)

In [10]:
df[['artist']].value_counts()


artist         
Porter Robinson    2059
Fred again..       2014
The Weeknd         1842
SVDDEN DEATH       1469
Skrillex           1468
                   ... 
tsubi club            1
trak-side             1
tothegood             1
tomcbumpz             1
tofubeats             1
Name: count, Length: 2523, dtype: int64

In [11]:
df[['track_id']].value_counts()

track_id                                                  
Out of Time                                                   296
Rumble                                                        260
What You Know Bout Love                                       204
Death Waltz                                                   188
Jungle                                                        176
                                                             ... 
(I Can't Get No) Satisfaction                                   1
&burn (with Vince Staples)                                      1
$uicideboy$ Were Better in 2015                                 1
#1 Fan                                                          1
"Was He Slow?" - Music From The Motion Picture Baby Driver      1
Name: count, Length: 8078, dtype: int64

Scope note: This notebook analyzes artist and track dominance using play counts only. Total listening time is not available in the current dataset and will not be inferred.

**Insights — Top Artists by Play Count**

- Observation: 2523 total different artists
- Interpretation: top 5 listed with plays per artist
- Implication: showing top artists in descending, Porter, Fred, Weeknd, SVDDEN DEATH, Skrillex

In [ ]:
**Insights — Top Songs by Play Count**

- Observation: 8078 total songs in data
- Interpretation: top 5 most top listened songs listed 
- Implication: showing top songs descending, out of time, rumble, what you know bout love, death waltz, jungle

In [18]:
df['track_id'].value_counts().head(10)


track_id
Out of Time                    296
Rumble                         260
What You Know Bout Love        204
Death Waltz                    188
Jungle                         176
Comprehension Barrier          171
Danielle (Smile On My Face)    166
adore u                        164
RATATA                         151
Places to Be                   148
Name: count, dtype: int64

In [22]:
df['artist'].value_counts().head(10) / len(df) 



artist
Porter Robinson    0.031151
Fred again..       0.030470
The Weeknd         0.027868
SVDDEN DEATH       0.022225
Skrillex           0.022209
Flume              0.020394
Young Thug         0.016854
Kanye West         0.016521
The Beatles        0.015190
Pop Smoke          0.014206
Name: count, dtype: float64

In [29]:
current = df['artist'].value_counts().head(10) / len(df) 
sum = current.sum()
print(f"{sum:.6%}")

21.708675%


### Artist Concentration

The top 10 artists account for approximately **21.708675%** of total listening activity. This indicates that listening behavior is moderately to highly concentrated, with a small group of artists receiving a disproportionate share of plays relative to the long tail of infrequent listens.
